src/doc/unstable-book/src/compiler-flags/codegen-backend.md

```diff
@@ -0,0 +1,28 @@
+# `codegen-backend`
+
+The tracking issue for this feature is: [#77933](https://github.com/rust-lang/rust/issues/77933).
+
+------------------------
+
+This feature allows you to specify a path to a dynamic library to use as rustc's
+code generation backend at runtime.
+
+Set the `-Zcodegen-backend=<path>` compiler flag to specify the location of the
+backend. The library must contain a function named `__rustc_codegen_backend`
+with a signature of `fn() -> Box<dyn rustc_codegen_ssa::traits::CodegenBackend>`.
```

Comment:
It must also have the `dylib` crate type and to prevent duplicate symbols messing up everything link against `rustc_driver`.